savant/deepstream/nvinfer/element_config.py

```diff
@@ -123,6 +123,20 @@ def process(self, msg, kwargs):
 
     model_config = OmegaConf.merge(model, model_config)
 
+    # try to parse engine file and check for a match
+    if model_config.engine_file:
+        parse_result = NvInferConfig.parse_model_engine_file(model_config.engine_file)
+        if parse_result and (
+            model_config.batch_size,
+            model_config.gpu_id,
+            model_config.precision,
+        ) != (
+            parse_result['batch_size'],
+            parse_result['gpu_id'],
+            parse_result['precision'],
+        ):
+            model_config.engine_file = None
```

Comment:
I think additional logging (at least on debug level) can be helpful here.

Reply:
Added logger.info